src/object-store/src/util.rs

```diff
@@ -123,6 +123,20 @@ pub fn normalize_path(path: &str) -> String {
     p
 }
 
+// This logical tries to extract parent path from the object storage operation
+// the function also relies on assumption that the region path is built from
+// pattern `<data|index>/catalog/schema/table_id/....`
+//
+// this implementation tries to extract at most 3 levels of parent path
+pub(crate) fn extract_parent_path(path: &str) -> &str {
+    // split the path into `catalog`, `schema` and others
+    path.char_indices()
+        .filter(|&(_, c)| c == '/')
+        // we get the data/catalog/schema from path, split at the 3rd /
+        .nth(2)
```

Comment:
The `2` means it should always maintain the layout:

```
<data_home>/<catalog>/<schema>/...
```

And the `region_dir` is `<catalog>/<schema>/...`. Actually, the `<data_home>` is not always one directory, maybe it can be:

```
data/<cluster>/<catalog>/<schema>/...
```

For this scenario, the magic number `2` will not work.

Reply:
The possible elegant solution may:

1. Make the extract function as the field in `PrometheusMetricsLayer` and be initialized by the caller;
2. The caller should always know how to cut the directory since it always knows how to organize the data;

Reply:
This is what I thought to be. But I tested by myself that `<data_home>` seems not included in this path, this is already a subpath when operating the storage. I placed a `dbg!()` call in read and it shows:

```
[src/object-store/src/layers/prometheus.rs:157:9] path = "data/greptime/public/1024/1024_0000000000/manifest/00000000000000000000.json"
```

The `data_home` is `/tmp/greptimedb`. 

Correct me if I missed something.

Reply:
We set the data home as the root path of the OpenDAL operator which is not included in the path. But I'm not sure about the cluster mode.

Reply:
You're right. I didn't read the specific logic of `new_s3_object_store()`. The [root](https://github.com/GreptimeTeam/greptimedb/blob/main/src/datanode/src/store/s3.rs#L35) is already applied for all the operations. It's OK for the original implementation.